File: problems/easy/Arrays/27.Remove Element/remove_element.py
from typing import List

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k = 0
        # O(n) time | O(1) space
        for i in range(len(nums)):
            if nums[i] != val:
                nums[i], nums[k] = nums[k], nums[i]
                k += 1
        return k
        

File: problems/easy/Arrays/27.Remove Element/remove_element.cpp
#include <vector>

class Solution {
public:
    // O(n) time | O(1) space
    int removeElement(std::vector<int>& nums, int val) {
        int k = 0;
        for(int i = 0; i < nums.size(); ++i) {
            if(nums[i] != val) {
                swap(nums, i, k);
                k += 1;
            }
        }
        return k;
    };

    void swap(std::vector<int>& nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }   
};